dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/importer/IdSchemeTests.java

```diff
@@ -0,0 +1,242 @@
+    {
+        String teiId = new IdGenerator().generateUniqueId();
+
+        TestData data = new TestData( idScheme );
+
+        EventDataBuilder eventDataBuilder = new EventDataBuilder()
+            .setOu( data.getOrgUnit() )
+            .setProgramStage( data.getTrackerProgramStage() )
+            .addDataValue( data.getDataElement(), DataGenerator.randomString() )
+            .setAssignedUser( Constants.SUPER_USER_ID )
+            .setProgram( data.getTrackerProgram() );
+
+        EnrollmentDataBuilder enrollmentDataBuilder = new EnrollmentDataBuilder()
+            .setOu( data.getOrgUnit() )
+            .setProgram( data.getTrackerProgram() )
+            .addEvent( eventDataBuilder );
+
+        JsonObject payload = new TeiDataBuilder()
+            .setId( teiId )
+            .setTeiType( data.getTrackedEntityType() )
+            .setOu( data.getOrgUnit() )
+            .addAttribute( data.trackedEntityAttribute, DataGenerator.randomString() )
+            .addEnrollment( enrollmentDataBuilder )
+            .addRelationship( new RelationshipDataBuilder()
+                .setRelationshipType( data.relationshipType )
+                .setFromTrackedEntity( this.createTei() )
+                .setToTrackedEntity( teiId ) )
+            .array();
+
+        trackerActions.postAndGetJobReport( payload,
+            new QueryParamsBuilder()
+                .addAll( "async=false", "idScheme=" + idScheme ) )
+            .validateSuccessfulImport();
+    }
+
+    @MethodSource( "provideIdSchemeArguments" )
+    @ParameterizedTest( name = "POST to /tracker?idScheme={0}" )
+    public void shouldImportEventProgramDataByIdScheme( String idScheme )
+    {
+        TestData data = new TestData( idScheme );
+
+        JsonObject object = new EventDataBuilder()
+            .setOu( data.getOrgUnit() )
+            .setProgramStage( data.getEventProgramStage() )
+            .setProgram( data.getEventProgram() )
+            .setAttributeCategoryOptions( Arrays.asList( data.categoryOption ) )
+            .array();
+
+        trackerActions
+            .postAndGetJobReport( object, new QueryParamsBuilder().addAll( "async=false", "idScheme=" + idScheme ) )
+            .validateSuccessfulImport();
+    }
+
+    @Test
+    public void shouldImportEventsWithDifferentProgramSchemes()
+    {
+        String programIdScheme = "CODE";
+        String ouIdScheme = "NAME";
+        String programStageIdScheme = "ATTRIBUTE:" + ATTRIBUTE_ID;
+        TestData data = new TestData();
+
+        JsonObject event = new EventDataBuilder()
+            .setProgram( new TestData( programIdScheme ).getEventProgram() )
+            .setProgramStage( new TestData( programStageIdScheme ).getEventProgramStage() )
+            .setOu( new TestData( ouIdScheme ).getOrgUnit() )
+            .setAttributeCategoryOptions( Arrays.asList( data.getCategoryOption() ) )
+            .array();
+
+        // act
```

Comment:
I don't think we need arrange, act, assert comments. You've grouped the code nicely, so its clear to me 😄